extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java

```diff
@@ -237,24 +237,26 @@ public void register(RegistrationContext registrationContext) {
                     configurator.addQualifier(REST_CLIENT);
                     final String configPrefix = computeConfigPrefix(restClientName.toString(), entry.getValue());
                     final ScopeInfo scope = computeDefaultScope(capabilities, config, entry, configPrefix);
-                    final Class<?> annotationProvider = checkAnnotationProviders(entry.getValue(),
-                            restClientAnnotationProvider.orElse(null));
-                    configurator.scope(scope);
-                    configurator.creator(m -> {
-                        // return new RestClientBase(proxyType, baseUri).create();
-                        ResultHandle interfaceHandle = m.loadClass(restClientName.toString());
-                        ResultHandle baseUriHandle = m.load(getAnnotationParameter(entry.getValue(), "baseUri"));
-                        ResultHandle configPrefixHandle = m.load(configPrefix);
-                        ResultHandle annotationProviderHandle = annotationProvider == null ? m.loadNull()
-                                : m.loadClass(annotationProvider);
-                        ResultHandle baseHandle = m.newInstance(
-                                MethodDescriptor.ofConstructor(RestClientBase.class, Class.class, String.class, String.class,
-                                        Class.class),
-                                interfaceHandle, baseUriHandle, configPrefixHandle, annotationProviderHandle);
-                        ResultHandle ret = m.invokeVirtualMethod(
-                                MethodDescriptor.ofMethod(RestClientBase.class, "create", Object.class), baseHandle);
-                        m.returnValue(ret);
-                    });
+                    for (RestClientAnnotationProviderBuildItem restClientAnnotationProvider : restClientAnnotationProviders.get()) {
```

Comment:
@VasilisAndritsoudis I don't think this code is correct, these providers represent only one parameter which needs to be passed to the underlying `RestClientBase`, so one can't iterate over them. Please restore the orginal code; you need to have this list converted correctly to `annotationProviderHandle`.

Reply:
@sberyozkin  Sorry about the delay, I had a look at it and I don't understand how it will work if we remove the `for` loop. This is because we have declared `List<RestClientAnnotationProviderBuildItem> restClientAnnotationProviders` as a `List`, which means if we remove the loop the List itself will be passed instead of the contained Providers. I can change it but I would like to get your opinion on this.